website/blog/2022-05-10-acf-in-a-project.md

```diff
@@ -0,0 +1,140 @@
+<!--truncate-->
+
+## Ways of registering ACF fields
+
+There are multiple approaches to register ACF fields. The easiest way is through WP Admin interface. But, if you're using Git, this approach isn't the best since the fields are being saved in the database, which means you would have to export and import fields into your staging and production environment each time you make a change.
+
+The recommended way is to register the fields with PHP. There is also an option to register the fields with **_.json_** files, but in Eightshift dev kit, we already have some goodies which will make registering fields with PHP a breeze.
+
+## Getting PHP code export
+
+The easiest way to get the PHP code that you will use for registering ACF fields is by adding those fields through the WP Admin ACF interface first. As an example, I'll add a field group called **_Intro_**, which will contain one text field of the same name. That field group will be displayed only on posts. You can add a few more fields, but for the sake of simplicity, I'll use only one field.
+
+When you set all the fields you need, save them and go to `Custom Fields -> Tools`. Here you will see the option to **_Export Field Groups_**. Simply select which field groups you want to export and select **_Generate PHP_**.
+
+![ACF PHP code export](/img/blog/acf-generate-php.png)
+
+This will generate a PHP code snippet that you can use in your theme. But now you may be wondering, where exactly should that code go?
+
+## CustomMeta class
+
+Those ACF goodies in Eightshift dev kit I talked about earlier? Let me introduce you to one of them. We have a WP CLI command which we can use to generate a CustomMeta class where we can add our field groups. The command is `wp boilerplate create_acf_meta`. This command has one required parameter, and that is `name`. To create a class that we will use for registering our custom fields, we'll use the following command:
+
+```bash
+wp boilerplate create_acf_meta --name=intro
+```
+
+This command will generate a **_CustomMeta_** folder inside **_src_** folder and add a new file called **_IntroAcfMeta.php_**. Inside that file, you should see the following method:
+```php
+public function fields(): void
+{
+	if (function_exists('acf_add_local_field_group')) {
+		\acf_add_local_field_group([]);
+	}
+}
+```
+
+Go back to your PHP code export for ACF fields and select everything inside the `acf_add_local_field_group()` function and paste it into your function. To make it in line with our coding standards, we have to do the following:
+- replace `array()` with `[]`
+- make every label translatable and escaped - use `esc_html__()`
+- instead of hardcoding the field name, replace it with a class constant
```

Comment:
Maybe explain why this is a good idea. (e.g. "to allow for easy access to field values" or "for better sustainability", etc.)